api-common-java/src/main/java/com/google/api/core/ApiFutures.java

```diff
@@ -159,57 +232,89 @@ public static <V, X> ApiFuture<X> transform(
             executor));
   }
 
+  /**
+   * Creates a new {@code ApiFuture} whose value is a list containing the values of all its input
+   * futures, if all succeed.
+   *
+   * <p>The list of results is in the same order as the input list.
+   *
+   * <p>Note that this method is a delegate of {@link Futures#allAsList(Iterable)}.
+   *
+   * @param futures Futures to combine
+   * @return A future that provides a list of the results of the component futures
+   * @see Futures#allAsList(Iterable)
+   */
   public static <V> ApiFuture<List<V>> allAsList(
       Iterable<? extends ApiFuture<? extends V>> futures) {
     return new ListenableFutureToApiFuture<>(
         Futures.allAsList(
             Iterables.transform(
                 futures,
-                new Function<ApiFuture<? extends V>, ListenableFuture<? extends V>>() {
-                  public ListenableFuture<? extends V> apply(ApiFuture<? extends V> apiFuture) {
-                    return listenableFutureForApiFuture(apiFuture);
-                  }
-                })));
+                (Function<ApiFuture<? extends V>, ListenableFuture<? extends V>>)
```

Comment:
Please remove the cast. It shouldn't be needed.

Reply:
This is outdated.